lib/iris/tests/unit/analysis/maths/test__inplace_common.py

```diff
@@ -0,0 +1,96 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the :func:`iris.analysis.maths._inplace_common` function."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+from iris.cube import Cube
+import numpy as np
+
+from iris.analysis.maths import _inplace_common
+
+
+class Test(tests.IrisTest):
+    def setUp(self):
+        self.scalar_int = 5
+        self.scalar_float = 5.5
+
+        self.float_data = np.array([8, 9], dtype=np.float64)
+        self.int_data = np.array([9, 8], dtype=np.int64)
+
+        self.float_cube = Cube(self.float_data)
+        self.int_cube = Cube(self.int_data)
+
+        self.op = 'addition'
+        self.emsg = 'Cannot perform inplace {}'.format(self.op)
+
+    def test_float_cubes(self):
+        result = _inplace_common(self.float_cube, self.float_cube, self.op)
+        self.assertIsNone(result)
```

Comment:
@bjlittle interestingly `_inplace_common` doesn't change anything anywhere, it's just a pass-through function that will either:

* do nothing (most cases) -- and so have an implicit return value of `None`, or
* fail with an error iff both cube and other have a dtype kind of `'i'`.

Further, I decided I wanted to explicitly check the known good cases, because if they become no longer good for some reason we want these tests to start failing.

But a banner comment on this slightly esoteric testing approach would probably be a help to our future selves!